Show field icons in default edit views

lukemelia opened the pull request:
- [x] design feedback
- [x] tests

![image](https://github.com/user-attachments/assets/a9dd2f48-48e0-4120-bdd4-93f450a5b0bc)

![image](https://github.com/user-attachments/assets/f3c7116e-1cef-4381-aa1a-ddd112037786)

lukemelia committed:
Show field icons in default edit views

jurgenwerk commented on packages/base/default-templates/isolated-and-edit.gts:
```diff
@@ -11,13 +13,21 @@ export default class DefaultCardDefTemplate extends GlimmerComponent<{
         {{#unless (eq key 'id')}}
+          {{#if (eq @format 'edit')}}
+            {{log (get @model key)}}
+            {{log (get (get (getField @model.constructor key) 'card') 'icon')}}
```
The logs are probably not needed anymore?

lukemelia committed:
Remove logging

lukemelia committed:
Add test coverage for edit format icons

lukemelia committed:
Test fixes

lukemelia committed:
Improve vertical alignment of edit labels and content

- edit formats no longer render embedded view in case of canEdit==false; instead they render read-only/disabled form editing

lukemelia committed:
Improve vertical alignment per design review

lukemelia committed:
Test updates

backspace reviewed (commented):
I like the addition of these icons, the alignment between them and the label text is chaotic to me though. They all look off-centre to varying degrees.

<img width="889" alt="8*) * 2024-10-17 16-30-06" src="https://github.com/user-attachments/assets/0d6ff3b9-048a-4652-b8ee-e019df3070a7">

is a pixel or so lower than

<img width="780" alt="8*) * 2024-10-17 16-31-09" src="https://github.com/user-attachments/assets/88033f9b-205e-4a27-a51d-f9af46483ebc">

It’s more pronounced on the smaller ones:

<img width="931" alt="8*) * 2024-10-17 16-31-49" src="https://github.com/user-attachments/assets/e5b6f791-e00c-43d4-8dc4-7d856896c4b9">

<img width="1060" alt="8*) * 2024-10-17 16-32-30" src="https://github.com/user-attachments/assets/f6922336-7400-41dd-a7a9-3c5042aeb963">

This seems like an unavoidable problem with icon sets and maybe per-icon tweaks are undesirable, but I think the smaller ones at least could all move up 1px.

lukemelia commented:
@backspace I agree, but don't want to start tweaking per icon. I'm going to merge as is and then we can review with design and adjust.
